### PR TITLE
feat(python): FirmwareUpdateData payload helper

### PR DESCRIPTION
**Pull request by amitlissack**
Add FirmwareUpdateData.create method that computes checksum and makes a FirmwareUpdateData instance.

### CODE REVIEW

File: python/opentrons_ot3_firmware/constants.py [hunk 1]
```diff
@@ -14,10 +14,10 @@ class NodeId(int, Enum):
...
     head_l = 0x51
     head_r = 0x52
-    pipette_bootloader = pipette | 0xf
-    gantry_x_bootloader = gantry_x | 0xf
-    gantry_y_bootloader = gantry_y | 0xf
-    head_bootloader = head | 0xf
+    pipette_bootloader = pipette | 0xF
+    gantry_x_bootloader = gantry_x | 0xF
+    gantry_y_bootloader = gantry_y | 0xF
+    head_bootloader = head | 0xF
```
Comment (DerekMaggio) [inline python/opentrons_ot3_firmware/constants.py:20]: Does this change anything by capitalizing it?

File: python/opentrons_ot3_firmware/messages/payloads.py [hunk 2]
```diff
@@ -183,6 +186,21 @@ def __post_init__(self) -> None:
...
+        """Create a firmware update data payload."""
+        checksum = 0
+        obj = FirmwareUpdateData(
+            address=utils.UInt32Field(address),
+            num_bytes=utils.UInt8Field(len(data)),
+            reserved=utils.UInt8Field(0),
+            data=FirmwareUpdateDataField(data),
+            checksum=utils.UInt16Field(checksum),
+        )
+        checksum = (1 + ~sum(obj.serialize())) & 0xFFFF
```
Comment (DerekMaggio) [inline python/opentrons_ot3_firmware/messages/payloads.py:200]: Why do you bitwise and this?

**Review by DerekMaggio — COMMENTED** [review]
Looks fine to me. But I don't understand enough what you are doing with the checksum to approve it

File: python/opentrons_ot3_firmware/constants.py (hunk 1 above)
Comment (sfoster1) [reply python/opentrons_ot3_firmware/constants.py:20]: nope

File: python/opentrons_ot3_firmware/messages/payloads.py (hunk 2 above)
Comment (sfoster1) [reply python/opentrons_ot3_firmware/messages/payloads.py:200]: it's a little clearer if you're used to reading these, if i don't see it i reflexively go "wait what happens on overflow"

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

File: python/opentrons_ot3_firmware/messages/payloads.py (hunk 2 above)
Comment (DerekMaggio) [reply python/opentrons_ot3_firmware/messages/payloads.py:200]: Also would `(0x0001  + ~sum(obj.serialize()))` be equivalent?

File: python/opentrons_ot3_firmware/constants.py (hunk 1 above)
Comment (amitlissack) [reply python/opentrons_ot3_firmware/constants.py:20]: Nah. Black just reformatted it.

File: python/opentrons_ot3_firmware/messages/payloads.py (hunk 2 above)
Comment (amitlissack) [reply python/opentrons_ot3_firmware/messages/payloads.py:200]: It's a common way to perform a checksum. Add the bytes, NOT them and add 1 (make it into a negative number).

`& 0xFFFF` is to restrict the result to 16 bits.

**Pull request merged**